packages/pelagos/src/treeView/TreeNode.js

```diff
@@ -145,15 +151,19 @@ const TreeNode = ({id, labelClassName, label, icon: Icon, expanded, loading, chi
 		[expanded, level, onSelect, onToggle, path, selected, setFocused]
 	);
 
+	if (!expanded && focused.length > childrenLevel && arrayEquals(focused.slice(0, childrenLevel), path)) {
```

Comment:
There are no references to any nodes, it will cause a re-render if the condition is true so I didn't want to wait until `useEffect` is called. It's ugly but I couldn't think of a better way to do it.